### PR TITLE
synchronize the project name between the frontend and the backend (workflow xml)

### PR DESCRIPTION
**Pull request by aminelouati**

### CODE REVIEW

**Review by amouhoub — APPROVED** [review]

File: src/main/java/org/ow2/proactive/catalog/service/WorkflowXmlManipulator.java [hunk 1]
```diff
@@ -111,6 +133,10 @@ private Element replaceJobName(Element element, String jobName) {
         return replaceAttributeValue(element, "name", jobName);
     }
 
+    private Element replaceOrAddProjectName(Element rootElement, String projectName) {
```
Comment (nebilbenmabrouk) [inline src/main/java/org/ow2/proactive/catalog/service/WorkflowXmlManipulator.java:136]: Check that projectName is an empty space :)

File: src/main/java/org/ow2/proactive/catalog/service/WorkflowXmlManipulator.java [hunk 2]
```diff
@@ -82,6 +82,28 @@
         }
     }
 
+    public byte[] replaceOrAddProjectNameOnJobLevel(final byte[] xmlWorkflow, String projectName) {
```
Comment (nebilbenmabrouk) [inline src/main/java/org/ow2/proactive/catalog/service/WorkflowXmlManipulator.java:85]: Check that projectName is an empty space, if it is empty, do nothing

Comment (aminelouati) [reply src/main/java/org/ow2/proactive/catalog/service/WorkflowXmlManipulator.java:85]: The projectName value (empty string or not) provided by the user will be synchronized in the underlying xml file.

File: src/main/java/org/ow2/proactive/catalog/service/WorkflowXmlManipulator.java (hunk 1 above)
Comment (aminelouati) [reply src/main/java/org/ow2/proactive/catalog/service/WorkflowXmlManipulator.java:136]: projectName can be empty string

File: src/main/java/org/ow2/proactive/catalog/service/WorkflowXmlManipulator.java (hunk 2 above)
Comment (fviale) [reply src/main/java/org/ow2/proactive/catalog/service/WorkflowXmlManipulator.java:85]: if the parameter projectName is empty, don't we expect to actually remove the attribute (this is the behavior in the studio) ?

Comment (aminelouati) [reply src/main/java/org/ow2/proactive/catalog/service/WorkflowXmlManipulator.java:85]: Okey. I will update the PR

**Review by fviale — APPROVED** [review]
LGTM, to be tested live :-)

**Pull request merged**